### ISSUE TITLE
Filebased ContentBlocks

### ISSUE THREAD

**Issue by AaronDB2** [Sterc/revolution#28]

**Comment by gpsietzema** [comment]
Please note: this will not be a PR towards the CB repo. It's proprietary software, written by the wonderful people of modmore: https://modmore.com/contentblocks/

This issue is part of the '21 MODX SnowUp/BugHunt and has a bounty on it (https://vi.to/hubs/modx-meetup).

**What is this about?**
What we and other agencies are looking for is a workflow/automated process which makes it easy to make ContentBlocks templates, fields and layouts filebased, so we can edit them in our favorite IDE and maintain them in our git-based version control systems.

Example of how we do it now (using a snippet to render blocks):
![image](https://user-images.githubusercontent.com/381017/109498659-102c0180-7a94-11eb-9c04-80d7f8ae3d20.png)

Downsides of this:
- Unable to export/import, because all the referenced templates/chunks-files are missing
- It's a hack and feels wrong ;-)

Requirements:
1. No PR, since it's kind of hard to PR this... A zipfile + reply to this issue will be sufficient
2. Please explain your workflow
3. If your solution comes with CLI scripts, they should run in a watcher. So the script has to detect if changes were made.

**Comment by Mark-H** [comment]
How's this? 

- https://modmore.com/blog/2021/contentblocks-v1.11/
- https://docs.modmore.com/en/ContentBlocks/v1.x/Fields/Templates.html#page_Using+%40FILE+templates

**Comment by sdrenth** [comment]
@Mark-H This sounds promising! Also glad to see a fix released for the export/import of repeater fields. I'm guessing the file based configuration of blocks is still something we can expect in V2? Great work!

**Comment by Mark-H** [comment]
Yeah a full built-in static workflow is still in the tracker for 2.0. 

Had an interesting chat with Ben "Mon Cheri" Davis earlier this week that prompted some interesting ideas, but I wont be able to actually work on that (and comment too much on those ideas publicly) until other parts of v2 are done. ;)

**Comment by Mark-H** [comment]
For obvious bounty-related reasons I'm curious if you've had a chance to try out the update to see to what extent that resolves the challenges Sterc was facing with a static workflow. ;)
